### PR TITLE
chore: 🆙 upgrade react 17

### PR DESCRIPTION
**Pull request by hengkx**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [x] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

https://github.com/ant-design/ant-design/issues/26136

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Support React 17. |
| 🇨🇳 Chinese | 支持 React 17。 |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

**Comment by hengkx** [comment]
@yoyo837  你怎么改图标的

**Comment by hengkx** [comment]
https://github.com/enzymejs/enzyme/issues/2429

**Comment by yoyo837** [comment]
> @yoyo837  你怎么改图标的

从别处复制的，输入法怎么打还不知道。😂

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/issues/26136

**Comment by afc163** [comment]
- [ ] bisheng 里还有 react 16 的依赖，导致网站构建失败
- [ ] react 16 相关的 test 要加一下

**Comment by yoyo837** [comment]
enzyme 支持后再更换吗？

**Comment by afc163** [comment]
对。

File: tests/setup.js [hunk 1]
```diff
@@ -31,7 +31,10 @@ if (typeof window !== 'undefined') {
 
 const Enzyme = require('enzyme');
 
-const Adapter = require('enzyme-adapter-react-16');
+const Adapter =
+  process.env.REACT === '16'
+    ? require('enzyme-adapter-react-16') // eslint-disable-line import/no-extraneous-dependencies,import/no-unresolved
```
Comment (ycjcl868) [inline tests/setup.js:36]: `enzyme-adapter-react-16` 依赖还是需要装

Comment (afc163) [reply tests/setup.js:36]: 有的，在 ci 里装了。

**Pull request merged**